pkg/customresource/customresource.go

```diff
@@ -157,10 +157,11 @@ func rawCRDFromFile(path string) ([]byte, error) {
 	return yamls.ReadFile(path)
 }
 
-func waitForCRDInit(context Context, resource CustomResource) error {
+func waitForCRDInit(crcontext Context, resource CustomResource) error {
 	crdName := fmt.Sprintf("%s.%s", resource.Plural, resource.Group)
-	return wait.Poll(context.Interval, context.Timeout, func() (bool, error) {
-		crd, err := context.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(contextpkg.TODO(), crdName, metav1.GetOptions{})
+	ctx := contextpkg.TODO()
+	return wait.PollUntilContextTimeout(ctx, crcontext.Interval, crcontext.Timeout, false, func(contextpkg.Context) (bool, error) {
+		crd, err := crcontext.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, crdName, metav1.GetOptions{})
```

Comment:
Why do we need to use `contextpkg.TODO()` in `wait.PollUntilContextTimeout()`. Why can't we use context received as a function argument? Do you see any issue here?

```suggestion
	return wait.PollUntilContextTimeout(context, context.Interval, context.Timeout, false, func(contextpkg.Context) (bool, error) {
		crd, err := crcontext.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(contextpkg.TODO(), crdName, metav1.GetOptions{})
```

Reply:
The context received from the function argument is of custom type:
```
type Context struct {
    Clientset             kubernetes.Interface
    APIExtensionClientset clientset.Interface
    Interval              time.Duration
    Timeout               time.Duration
}
```
But inside the function, requirement is for native `context.Context`

Reply:
I see, we should get context of context.Context type from parent function and use it here.
E.g if the parent context get cancelled, this function should also get terminated

Reply:
Done